app/src/main/java/org/fossasia/openevent/general/event/EventDao.kt

```diff
@@ -26,6 +27,9 @@ interface EventDao {
     @Query("UPDATE Event SET favorite = :favorite WHERE id = :eventId")
     fun setFavorite(eventId: Long, favorite: Boolean)
 
+    @Query("SELECT id from Event WHERE favorite = 1")
```

Comment:
You only want events which are within a range.

Suppose you have 1000 events and networks only loads 23 ,45, 67, 89 ID events, you don't want to deal with all thousand favorite events. Refer back to my query